### ISSUE TITLE
Halted VM cannot be destroyed

### ISSUE THREAD

**Issue by daFritz84** [myoung34/vagrant-ovirt4#124]
My understanding is that I can stop a VM with vagrant and then destroy it. Using this provider I get an error as shown in the log below. Bringing the machine up and then destroying it works OK.

@myoung34 Is this a bug or intended behaviour. In case this is a bug I will look into fixing it. Please advise.

log:
```
❯ vagrant halt
==> default: Halting VM...
==> default: Waiting for VM to shutdown...
❯ vagrant destroy
==> default: Halting VM...
Guest-specific operations were attempted on a machine that is not
ready for guest communication. This should not happen and a bug
should be reported.
❯ vagrant up
Bringing machine 'default' up with 'ovirt4' provider...
==> default: Starting VM.
==> default: Waiting for VM to get an IP address...
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Got IP: 10.20.190.70
==> default: Machine is booted and ready for use!
==> default: Rsyncing folder: /mnt/c/Users/SES/Source/ansible-icinga/ => /vagrant
❯ vagrant destroy
==> default: Halting VM...
==> default: Waiting for VM to shutdown...
==> default: Removing VM...
```

**Comment by daFritz84** [comment]
Looking through the open pull-requests, I think #116 might fix this behavior already.

**Comment by myoung34** [comment]
I merged #116 if you want to see if it fixes it

**Comment by daFritz84** [comment]
#116 worked beautifully.

An improved approach would be to check if the machine is already halted. I will look into that later, so I keep this issue open for now.

Log:
```
❯ vagrant halt
==> default: Halting VM...
==> default: Waiting for VM to shutdown...
❯ vagrant destroy
==> default: Halting VM...
==> default: Failed to shutdown guest gracefully.
==> default: Waiting for VM to shutdown...
==> default: Removing VM...
```

Tested with ovirt 4.4.2.6-1.el8 and Ubuntu 20.04 WSL on Windows 20H2

**Comment by myoung34** [comment]
Resolved by #125
